lib/graphql/static_validation/rules/mutation_root_exists.rb

```diff
@@ -0,0 +1,20 @@
+module GraphQL
+  module StaticValidation
+    class MutationRootExists
+      include GraphQL::StaticValidation::Message::MessageHelper
+
+      def validate(context)
+        return if context.schema.mutation
+
+        visitor = context.visitor
+
+        visitor[GraphQL::Language::Nodes::OperationDefinition].enter << -> (ast_node, prev_ast_node) {
+          if ast_node.operation_type == 'mutation' && !context.schema.mutation
```

Comment:
Is `!context.schema.mutation` already known at this point? (I think it would have caused an early return above.) Technically, someone could mess with the schema in the middle of validation but ... I don't think we need to support that case! (Same for `subscription` below, sorry for missing this earlier!)